bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/welcome/NAWelcomeHomeHandler.java

```diff
@@ -83,11 +82,14 @@ protected NAWelcomeHome updateReadings() {
                     }
                 });
 
+                Optional<NAWelcomeEvent> previousLastEvent = lastEvent;
                 result.getEvents().forEach(event -> {
-                    if (lastEvent == null || lastEvent.getTime() < event.getTime()) {
-                        lastEvent = event;
+                    if (!lastEvent.isPresent() || lastEvent.get().getTime() < event.getTime()) {
+                        lastEvent = Optional.of(event);
                     }
                 });
```

Comment:
This can be simplified to: `lastEvent = result.getEvents().stream().min((e1, e2) -> e1.getTime() - e2.getTime());`

Reply:
done